### PR TITLE
[MERGE] *: Migrate Bootstrap to version 5.1.3

### PR DESCRIPTION
**Pull request by rfr-odoo**
Migrate Bootstrap to version 5.1.3

some highlights of BS5 changes:
- Browser support: Internet Explorer is definitely gone (Goodbye! We won't miss you) ; not a huge change on the backend as we didn't supported it anymore anyway.
- Bootstrap's Javascript components doesn't use jQuery anymore! A compatibility layer is present but with some quirks, so avoid to use it if possible ; less jQuery is less overhead!
- The color contrast system in BS5 relies directly on WCAG 2.0 contrast algorithm (https://www.w3.org/TR/WCAG20-TECHS/G18.html). Some visual elements (like texts/buttons/...) may have a slightly different contrast than before.
- Background color classes ('bg-') only set the background color and don't change the foreground (aka. text) color anymore. We backported the '.text-bg-' from Bootstrap 5.2 to restore the previous behavior ; or use separate background ('.bg-') and foreground ('.text-') depending on your usecase.
- Media Breakpoint's usage has changed
- More and more CSS utility classes are introduced: use them in your templates instead of adding more custom CSS rules.
- Bootstrap still provides a quite powerful grid system, but please use it as intended by the framework (I'm looking at you '.col' without its parent '.row'). Also while keeping the same API, it was internally changed ; the main issue that you can encounter is that columns no longer apply 'position: relative', which could be solved by adding the '.position-relative' class when needed.
- BS5 doesn't allow to use multiple components on the same element (ie. adding a tooltip on a dropdown). It generates this kind of error: "Bootstrap doesn't allow more than one instance per element. Bound instance: bs.tooltip" (cf. https://github.com/twbs/bootstrap/issues/33940#issuecomment-838454777).
- Badge usage has changed a bit: '.badge' still exist but is now extended using utility classes (text and background colors, border-radius for pills) instead of specific '.badge-*' classes.
- '.input-group-{append,prepend}' are gone! You can now just add buttons and '.input-group-text' as direct children of the '.input-group'.
- Popover/alert components' close button use the new class '.btn-close'.
- Dropdown: the '.show' class is not added anymore to the parent group of dropdown ; some CSS selectors may still need some love.
- All data attributes of BS5 are now prefixed by 'bs-' (ie. 'data-toggle' is now 'data-bs-toggle').

Bootstrap documentation
https://getbootstrap.com/docs/5.1/migration

Task ID: 2766483

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by adr-odoo** [comment]
@beledouxdenis

**Review by AntoineVDV — CHANGES_REQUESTED** [review]

File: addons/payment/views/payment_templates.xml [hunk 1]
```diff
@@ -199,7 +199,7 @@
                             </span>
                             <!-- === "Test Mode" badge === -->
                             <span t-if="acquirer.state == 'test'"
-                                  class="badge-pill badge-warning"
+                                  class="rounded-pill text-bg-warning"
```
[the remark this answers is not in the document]
Comment (pparidans) [reply addons/payment/views/payment_templates.xml:202]: Same as for the previous comment.

File: addons/payment/views/payment_templates.xml [hunk 2]
```diff
@@ -63,12 +63,12 @@
...
                             <!-- === "Test Mode" badge === -->
                             <span t-if="acquirer.state == 'test'"
-                                  class="badge-pill badge-warning ml-1">
+                                  class="rounded-pill text-bg-warning ms-1">
                                 Test Mode
                             </span>
                             <!-- === Extra fees badge === -->
                             <t t-if="fees_by_acquirer.get(acquirer)">
-                                <span class="badge-pill badge-secondary ml-1">
+                                <span class="rounded-pill text-bg-secondary ms-1">
```
[the remark this answers is not in the document]
Comment (pparidans) [reply addons/payment/views/payment_templates.xml:71]: Actually the "badge" class was already missing... (cf. https://getbootstrap.com/docs/4.3/components/badge/)

[the remark this answers is not in the document]
Comment (AntoineVDV) [reply addons/payment/views/payment_templates.xml:71]: I'm fine with whichever adaption you need to do to the structure, classes, etc. to restore the original visual. If you correct inconsistencies while doing so, even better!

**Comment by xmo-odoo** [comment]
@robodoo override=ci/security

Bootstrap updates

**Review by adr-odoo — APPROVED** [review]
@robodoo rebase-merge p=1 r+

**Comment by d-fence** [comment]
robodoo p=0

**Comment by d-fence** [comment]
robodoo r+ p=0

**Comment by adr-odoo** [comment]
> And where is the upgrade script?

We are currently working on it with the website team.

**Comment by KangOl** [comment]
But it should have been merged with this PR.

**Comment by adr-odoo** [comment]
Personal note: +32,205 -27,824

**Comment by xmo-odoo** [comment]
> Personal note: +32,205 -27,824

Is that where you hid your treasure?

**Pull request closed without merging**